### ISSUE TITLE
[themes] Regression in transpiling Yarn workspaces

### ISSUE THREAD

**Issue by jlengstorf** [gatsbyjs/gatsby#15460]
## Description

On @ChristopherBiscardi’s stream today, he set up a local copy of `gatsby-theme-blog` and the Babel transpilation didn't work. It looks like something happened that is excluding local workspaces from the Babel transpilation.

For the next two weeks, you can watch the issue happen on https://www.twitch.tv/chrisbiscardi in the videos section.

### Steps to reproduce

1. Create a new site from the Gatsby blog theme starter
2. Pull a local copy of `gatsby-theme-blog`
3. Set it up as a workspace
4. On starting `develop` mode, it'll crash with "Unexpected token" errors.

### Expected result

Transpilation should happen.

### Actual result

Plugin for ES6 compilation is required to get `develop` to run.

**Comment by polishedwp** [comment]
This is interesting. 
I started with the `gatsby-starter-blog-theme` and it worked mostly fine. I can start up without any issues. 

I did notice an error message in the develop script  #15436, but it starts and works fine.

**Comment by ChristopherBiscardi** [comment]
Note that if you run into this in a workspaces environment, you can add this to your site's gatsby-config for now while we work out a fix.

```js
{
  resolve: `gatsby-plugin-compile-es6-packages`,
  options: {
    modules: [`gatsby-theme-blog`]
  }
},
```

Also @jlengstorf I'm an affiliate on Twitch so I think my videos get archived for 60 days now and here's a link to the specific video: https://www.twitch.tv/videos/448723773##

**Comment by sidharthachatterjee** [comment]
Fix published in `gatsby@2.13.9`
